velox/functions/lib/MapFromEntries.cpp

```diff
@@ -208,26 +211,36 @@ class MapFromEntriesFunction : public exec::VectorFunction {
           nulls, indices, decodedRowVector->size(), rowVector->childAt(1));
     }
 
-    // To avoid creating new buffers, we try to reuse the input's buffers
-    // as many as possible.
+    // For Presto, need construct map vector based on input nulls for possible
+    // outer expression like try(). For Spark, use the updated nulls unless it's
+    // empty.
+    if (throwOnNull_ || decodedRowVector->size() == 0) {
+      nulls = inputArray->nulls();
+    }
     auto mapVector = std::make_shared<MapVector>(
         context.pool(),
         outputType,
-        inputArray->nulls(),
+        nulls,
         rows.end(),
         inputArray->offsets(),
         sizes,
         wrappedKeys,
         wrappedValues);
 
-    checkDuplicateKeys(mapVector, *remianingRows, context);
+    checkDuplicateKeys(mapVector, *remainingRows, context);
     return mapVector;
   }
+
+  // If true, throws exception for null input or null entry (Presto behavior).
+  // Otherwise, returns null for these cases (Spark behavior).
```

Comment:
The `null input` and `null entry` seems unclear to me. Perhaps revise the comment a bit:

> If true, throws exception when input array is NULL or contains NULL entry (Presto's behavior). Otherwise, returns NULL (Spark's behavior).